Gems/Atom/RHI/DX12/Code/Source/RHI/RayTracingBlas.cpp

```diff
@@ -103,5 +103,26 @@ namespace AZ
 #endif
             return RHI::ResultCode::Success;
         }
+
+        D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAGS RayTracingBlas::GetAccelerationStructureBuildFlags(const RHI::RayTracingAccelerationStructureBuildFlags &buildFlags)
+        {
+            D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAGS dxBuildFlags = {};
+            if (RHI::CheckBitsAny(buildFlags, RHI::RayTracingAccelerationStructureBuildFlags::FAST_TRACE))
+            {
+                dxBuildFlags |= D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_PREFER_FAST_TRACE;
+            }
+
+            if (RHI::CheckBitsAny(buildFlags, RHI::RayTracingAccelerationStructureBuildFlags::FAST_BUILD))
+            {
+                dxBuildFlags |= D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_PREFER_FAST_BUILD;
+            }
+
+            if (RHI::CheckBitsAny(buildFlags, RHI::RayTracingAccelerationStructureBuildFlags::ENABLE_UPDATE))
+            {
+                dxBuildFlags |= D3D12_RAYTRACING_ACCELERATION_STRUCTURE_BUILD_FLAG_ALLOW_UPDATE;
+            }
+
+            return dxBuildFlags;
+        }
```

Comment:
This needs to be set within an `#ifdef AZ_DX12_DXR_SUPPORT`.

Reply:
Please fix this in a way that it will also compile.

Reply:
I have moved the macro up a line now to encompass the entire function. This should exclude it entirely when DX12 is not supported and not cause any compile errors.